openmc/mgxs/groups.py

```diff
@@ -0,0 +1,297 @@
+from collections.abc import Iterable
+from numbers import Real
+import copy
+import sys
+
+import numpy as np
+
+import openmc.checkvalue as cv
+
+
+class EnergyGroups(object):
+    """An energy groups structure used for multi-group cross-sections.
+
+    Parameters
+    ----------
+    group_edges : Iterable of Real
+        The energy group boundaries [eV]
+
+    Attributes
+    ----------
+    group_edges : Iterable of Real
+        The energy group boundaries [eV]
+    num_groups : int
+        The number of energy groups
+
+    """
+
+    def __init__(self, group_edges=None):
+        self._group_edges = None
+
+        if group_edges is not None:
+            self.group_edges = group_edges
+
+    def __deepcopy__(self, memo):
+        existing = memo.get(id(self))
+
+        # If this is the first time we have tried to copy object, create copy
+        if existing is None:
+            clone = type(self).__new__(type(self))
+            clone._group_edges = copy.deepcopy(self.group_edges, memo)
+
+            memo[id(self)] = clone
+
+            return clone
+
+        # If this object has been copied before, return the first copy made
+        else:
+            return existing
+
+    def __eq__(self, other):
+        if not isinstance(other, EnergyGroups):
+            return False
+        elif self.num_groups != other.num_groups:
+            return False
+        elif np.allclose(self.group_edges, other.group_edges):
+            return True
+        else:
+            return False
+
+    def __ne__(self, other):
```

Comment:
Is there a reason to override this method? From https://docs.python.org/3/reference/datamodel.html#object.__ge__:

By default, `__ne__()` delegates to `__eq__()` and inverts the result unless it is `NotImplemented`.